tools/elfdeps.cc

```diff
@@ -96,27 +94,111 @@ static const char *mkmarker(GElf_Ehdr *ehdr)
     return marker;
 }
 
-static void addDep(ARGV_t *deps,
-		   const char *soname, const char *ver, const char *marker)
+using elfmap = std::unordered_map<uint32_t, std::string>;
+
+static const elfmap archmap = {
+    { EM_NONE,		"none" },
+    { EM_386,		"x86" },
+    { EM_AARCH64,	"aarch" },
+    { EM_ALPHA,		"alpha" },
+    { EM_ARM,		"arm" },
+    { EM_BPF,		"bpf" },
+    { EM_FAKE_ALPHA,	"alpha" },
+    { EM_MIPS,		"mips" },
+    { EM_PPC,		"ppc" },
+    { EM_PPC64,		"ppc" },
+    { EM_RISCV,		"riscv" },
+    { EM_S390,		"s390" },
+    { EM_SPARC,		"sparc" },
+    { EM_SPARCV9,	"sparcv9" },
+    { EM_X86_64,	"x86" },
+};
+
+static const elfmap bitsmap = {
+    { ELFCLASSNONE,	"invalid" },
+    { ELFCLASS32,	"32" },
+    { ELFCLASS64,	"64" },
+};
+
+static const elfmap endianmap = {
+    { ELFDATANONE,	"invalid" },
+    { ELFDATA2LSB,	"l" },
+    { ELFDATA2MSB,	"b" },
+};
+
+static std::string elfname(const elfmap & emap, uint32_t val)
+{
+    auto res = emap.find(val);
+    return res != emap.end() ? res->second : "unknown";
+}
+
+static void armflags(GElf_Ehdr *ehdr, std::string & flags)
+{
+    if ((ehdr->e_flags & EF_ARM_ABI_FLOAT_HARD) == EF_ARM_ABI_FLOAT_HARD)
+	flags += "h";
+    if ((ehdr->e_flags & EF_ARM_ABI_FLOAT_SOFT) == EF_ARM_ABI_FLOAT_SOFT)
+	flags += "s";
```

Comment:
Likewise `hf` and `sf` here.

Reply:
The point with the flags field is that they're all single-letter items.